src/main/java/com/tstepnik/planner/service/TaskService.java

```diff
@@ -41,9 +44,16 @@ public Task getTask(Long taskId) {
 
     public Task addTask(Task task) {
         User user = authService.getLoggedUser();
+        ZonedDateTime now = ZonedDateTime.now(ZoneId.of("UTC")).truncatedTo(ChronoUnit.SECONDS);
+        task.setCreationDate(now);
         if (task.getImportance() == null) {
             task.setImportance(DEFAULT_IMPORTANCE);
         }
+        if (task.getPlannedFor() == null) {
+            LocalTime endOfTheDay = LocalTime.of(23, 59, 59);
```

Comment:
why are you setting plannedFor to the end of the day?

Reply:
All of this date code seems a little bit too complicated. It should be as simple as setting planned date to today

Reply:
I did it a little bit less complicated(i think😅) but i didn't find other way to set UTC time zone for hour and minutes than without ZonedDateTime. If I did the same code with LocalDateTime in database i got UTC time but it's compared to LocalTime so if i set midnight in database i got 22:00. With Zoned.. i got valid time.

Reply:
I mean that comparison with LocalDateTime seems to be right since users of your app might be all over the world.